### ISSUE TITLE
mypy plugins installed in project don't work

### ISSUE THREAD

**Issue by vrslev** [python-lsp/pylsp-mypy#90]
Hi! Thanks for your work.

I couldn't make mypy plugins to work since python-lsp-server is separate from specific project: they would have to be installed in the same environment as python-lsp-server. Do you have some advice?

**Comment by Richardk2n** [comment]
If `mypy` is on PATH, that is the preferred one for `pylsp-mypy`.
This means you can:
- Create a venv
- Install mypy and your desired plugins
- Activate the venv
- Start your IDE/pylsp server from your base environment
- Rejoice

**Comment by vrslev** [comment]
Is there a way to specify static path that `pylsp-mypy` uses to find `mypy` instead of PATH? In my case, it's always `./.venv/bin/mypy`

**Comment by Richardk2n** [comment]
No, it goes for `PATH` first and if that does not exist it uses the one installed in the environment in which `pylsp-mypy` is installed.
Depending on your use case, you could have a short script wrapping your IDE/pylsp execution to prepend the appropriate path of the cwd to `PATH` for this execution only.

**Comment by vrslev** [comment]
I’ll try it later, thanks! 😊 
I’m trying to move from VSCode to Zed and *this* is the thing I’m stuck currently

**Comment by Richardk2n** [comment]
So, you got it working?

**Comment by vrslev** [comment]
Zed doesn’t provide a way to customize PATH dynamically, or I didn’t find it. So, my use case can’t fulfilled, sadly. #91 would be I shortcut, but I understand that you have concerns about security

**Comment by Richardk2n** [comment]
Why does Zed need to do this?
You have multiple options for a wrapper script (prepend to `PATH` or activate the venv). If you don't want to write one, this works:
> If `mypy` is on PATH, that is the preferred one for `pylsp-mypy`. This means you can:
> 
> - Create a venv
> - Install mypy and your desired plugins
> - Activate the venv
> - Start your IDE/pylsp server from your base environment
> - Rejoice

It is a tiny bit manual, but your use case works.

**Comment by vrslev** [comment]
> It is a tiny bit manual, but your use case works.

This would work only if I open the editor from terminal. 90% of people won’t do that.

 I guess, wrapper script might work. I’ll check this later, thanks 👍

**Comment by vrslev** [comment]
Made it work with wrapper script: [dmypy](https://github.com/vrslev/dotfiles/blob/db9dcc3aee9a7d4f0c490b585d81d3391053bb41/bin/dmypy), [mypy](https://github.com/vrslev/dotfiles/blob/db9dcc3aee9a7d4f0c490b585d81d3391053bb41/bin/mypy), thanks for the advice 👍
